# COMET Comparison

The [COMET](https://www.cometadata.org/) project is an effort to build tools for collaboratively improving scholarly metadata. It is predicated on the idea that academic institutions can help improve metadata that is being made available by Crossref as part of DOI registration by publishers.

This notebook explores how our data, which aggregates information from Dimensions, Web of Science, Open Alex and PubMed differs from the metadata that is available from Crossref.

Since we are storing the verbatim metadata from each source, we can look for where this data differs substantially from Crossref. The goal is to find 10 records that illustrate how we could provide enhanced metadata to COMET.

In [73]:
import os
import dotenv
import pandas
import sqlalchemy

Set some variables and use them to connect to the database after you have [established an SSH tunnel](https://github.com/sul-dlss/rialto-airflow/wiki/Querying-with-pgAdmin) to the database server.

In [74]:
dotenv.load_dotenv()

db_password = os.environ.get("DB_PASSWORD_STAGE")
db_name = 'rialto_20250711222454'
db_user = 'analyst'

engine = sqlalchemy.create_engine(f'postgresql://{db_user}:{db_password}@localhost:9999/{db_name}')

In [75]:
pandas.set_option('display.max_colwidth', 0)
pandas.set_option('display.max_rows', 1000)

pandas.read_sql('select doi from publication where doi is not null limit 15', con=engine)

doi
0   10.1001/archneur.1981.00510110050006
1   10.1001/archneur.1982.00510160028005
2   10.1001/archneur.1983.04050020055011
3   10.1001/archneur.1985.04060020033010
4   10.1001/jamanetworkopen.2019.11519  
5   10.1001/jamanetworkopen.2020.29540  
6   10.1001/jamanetworkopen.2023.3640   
7   10.1001/jamanetworkopen.2023.3646   
8   10.1001/jamanetworkopen.2024.1828   
9   10.1001/jamaoncol.2017.5257         
10  10.1002/2327-6924.12532             
11  10.1002/9780470015902.a0004110      
12  10.1002/acn3.703                    
13  10.1002/acr2.11573                  
14  10.1002/adem.202300336

## Title

Compare the title we've extracted from SulPub, Dimensions, OpenAlex and WoS with what is in Crossref.

In [76]:
pandas.read_sql("""
  SELECT doi, title, crossref_title
  FROM (
      SELECT
          doi,
          title,
          crossref_json -> 'title' ->> 0 AS crossref_title
      FROM publication
      WHERE crossref_json IS NOT NULL
  )
  WHERE
      RTRIM(LOWER(title), '.') != RTRIM(LOWER(crossref_title), '.')
      AND crossref_title !~ '<'
  LIMIT 25
""", con=engine)

doi  \
0   10.1016/j.echo.2019.04.002          
1   10.1002/cncr.25881                  
2   10.1002/ski2.235                    
3   10.1027/1618-3169/a000578           
4   10.1158/1078-0432.ccr-24-2814       
5   10.1161/circulationaha.112.000343   
6   10.1016/j.jacc.2021.02.023          
7   10.1016/j.healun.2024.02.1179       
8   10.2147/pgpm.s201276                
9   10.1021/jm00110a010                 
10  10.1038/s41598-020-76344-3          
11  10.1002/stem.1422                   
12  10.1016/j.ajt.2023.02.014           
13  10.1021/bi010122j                   
14  10.1158/1535-7163.mct-22-0191       
15  10.1016/j.jcms.2021.02.003          
16  10.1007/s11307-017-1105-7           
17  10.1148/radiol.241051               
18  10.1161/circresaha.117.310584       
19  10.20944/preprints202403.0511.v1    
20  10.1161/circresaha.114.302884       
21  10.1016/j.healun.2024.02.1184       
22  10.1007/s00467-024-06438-x          
23  10.1016/j.healun.2024.02.1202       
24  10.1016/j.healun.2025.02.1028       

                                                                                                                                                                                                             title  \
0   2019 ACC/AHA/ASE Advanced Training Statement on Echocardiography (Revision of the 2003 ACC/AHA Clinical Competence Statement on Echocardiography): A Report of the ACC Competency Management Committee           
1   High-Intensity Focused Ultrasound (HIFU) Is Not Indicated for Treatment of Primary Bone Sarcomas                                                                                                                 
2   Development of a digital tool for home‐based monitoring of skin disease for older adults                                                                                                                         
3   Does Watching Videos With Natural Scenery Restore Attentional Resources?                                                                                                                                         
4   First-in-Human Clinical Trial of a Small-Molecule EBNA1 Inhibitor, VK-2019, in Patients with Epstein-Barr-Positive Nasopharyngeal Cancer, with Pharmacokinetic and Pharmacodynamic Studies.                      
5   Sustained Release of Engineered Stromal Cell–Derived Factor 1-&agr; From Injectable Hydrogels Effectively Recruits Endothelial Progenitor Cells and Preserves Ventricular Function After Myocardial Infarction   
6   COVID-19 in Adults With Congenital Heart Disease                                                                                                                                                                 
7   (534) Failure to Rescue in Heart Lung Transplantation: Progress Over 30 Years                                                                                                                                    
8   Pharmacogenetics of Pediatric Asthma: Current Perspectives                                                                                                                                                       
9   Topographically designed analogues of [D-Pen,D-Pen5]enkephalin.                                                                                                                                                  
10  Epstein–Barr virus peptides derived from latent cycle proteins alter NKG2A + NK cell effector function                                                                                                           
11  Natural killer cell‐activating receptor NKG2D mediates innate immune targeting of allogeneic neural progenitor cell grafts                                                                                       
12  Original Mutations in latent membrane protein 1 of Epstein-Barr virus are associated with increased risk of posttransplant lymphoproliferative disorder in children                  

The order of preference for title is:

* sulpub
* Dimensions
* OpenAlex
* Web of Science

Maybe title isn't the greatest place to start comparing since Crossref actually seems to be better in places. For example:

> (534) Failure to Rescue in Heart Lung Transplantation: Progress Over 30 Years

compared with Crossref:

> Failure to Rescue in Heart Lung Transplantation: Progress Over 30 Years

The published [article](https://doi.org/10.1016/j.jacc.2021.02.023) title doesn't have "(534)" in it.

> Original Mutations in latent membrane protein 1 of Epstein-Barr virus are associated with increased risk of posttransplant lymphoproliferative disorder in children

compared with Crossref:

> Mutations in latent membrane protein 1 of Epstein-Barr virus are associated with increased risk of posttransplant lymphoproliferative disorder in children`

The published [article](https://doi.org/10.1016/j.ajt.2023.02.014) title doesn't have "Original" in it.
    
It might be better to focus on areas where data we've collected isn't available from Crossref.

## Crossref Data

Examining a Crossref record like [this journal article](https://api.crossref.org/works/10.1098/rsos.150114) we can see that it includes metadata for:

* DOI
* title
* short title
* subtitle
* publisher
* journal title
* volume
* issue
* pages
* abstract
* publisher website URL
* article URL
* PDF URL
* publication date
* publication type
* record creation date
* update
* reference count
* author names
* author affiliations: e.g. `From the Departments of Cardiothoracic Surgery (A.N.S., J.W.M., Y.J.W.) and Bioengineering (A.N.S., Y.J.W.), Stanford University, CA.`
* references as DOIs
* language
* deposit date
* issued date
* ISSNs

As part of our workflow processing we are extracting a few pieces of information that is not available here:

* Article Processing Charge (APC)
* Open Access Classification
* Author ORCID
* Author email
* ROR for school
* Funder
* Federal Funder

Maybe that's enough? We can serialize the records we have as JSON for sharing.

## APC

In [77]:
pandas.read_sql("""
  SELECT doi, apc 
  FROM publication
  WHERE doi IS NOT NULL
    AND apc IS NOT NULL
""", con=engine)

doi    apc
0     10.1002/pbc.24328           4330 
1     10.1038/s41586-022-04489-4  11690
2     10.1002/ange.202101644      0    
3     10.1016/j.jim.2020.112936   0    
4     10.1002/art.42099           4940 
...                 ...            ... 
1384  10.7554/elife.93013         2450 
1385  10.7189/jogh.10.020380      2450 
1386  10.7189/jogh.12.04014       1200 
1387  10.7759/cureus.42051        2450 
1388  10.7759/cureus.14545        2450 

[1389 rows x 2 columns]

Here's a function that will output a publication with its authors and funders as JSON.

In [128]:
import json

# there's probably a better way to do this, but this will work for these purposes
def write_json(doi, filename):
    pub = pandas.read_sql('SELECT * from publication where doi = %(doi)s', params={'doi': doi}, con=engine).to_dict(orient='records')[0]

    pub['authors'] = pandas.read_sql(
        f'''
        SELECT *
        FROM author, pub_author_association
        WHERE author.id = pub_author_association.author_id
          AND pub_author_association.publication_id = {pub['id']}
        ''',
        con=engine).to_dict(orient='records')

    pub['funders'] = pandas.read_sql(
        f'''
        SELECT *
        FROM funder, pub_funder_association
        WHERE funder.id = pub_funder_association.funder_id
          AND pub_funder_association.publication_id = {pub['id']}
        ''',
        con=engine).to_dict(orient='records')
       
    json.dump(pub, open(f'data/comet/{filename}', 'w'), indent=2, sort_keys=True, default=str)

write_json('10.1038/s41586-022-04489-4', 'apc.json')
    

In [130]:
## Open Access

In [125]:
pandas.read_sql("""
  SELECT doi, open_access 
  FROM publication
  WHERE doi IS NOT NULL
    AND open_access IS NOT NULL
  LIMIT 100
""", con=engine)

doi open_access
0   10.1002/pbc.24328                     green     
1   10.1016/j.jamcollsurg.2012.06.133     closed    
2   10.1021/acsabm.0c00348                closed    
3   10.1038/s41586-022-04489-4            hybrid    
4   10.1021/acsabm.9b00769                closed    
5   10.1002/ange.202101644                closed    
6   10.1016/j.jim.2020.112936             closed    
7   10.1021/acs.nanolett.3c04172          green     
8   10.1002/art.42099                     bronze    
9   10.1002/aet2.11043                    bronze    
10  10.1002/cncr.23860                    bronze    
11  10.1002/ejoc.201800416                closed    
12  10.1002/marc.201900414                green     
13  10.1001/archneur.1981.00510110050006  closed    
14  10.1007/978-3-030-57193-1_22          closed    
15  10.1016/j.jpainsymman.2020.08.029     hybrid    
16  10.1016/s2152-2650(22)01453-7         closed    
17  10.3390/pathogens11101176             gold      
18  10.1016/j.fertnstert.2025.05.005      bronze    
19  10.1002/adma.202311949                hybrid    
20  10.1016/j.jaccas.2024.102715          gold      
21  10.1016/j.jacc.2019.01.069            hybrid    
22  10.1002/cnm.3869                      hybrid    
23  10.1007/s13239-024-00764-9            closed    
24  10.1021/acsaelm.0c00775               hybrid    
25  10.1093/abm/kaaf037                   closed    
26  10.1016/j.elspec.2021.147093          hybrid    
27  10.1021/acs.inorgchem.6b01779         closed    
28  10.1038/s41467-023-40611-4            gold      
29  10.1200/jco.2025.43.16_suppl.e24088   closed    
30  10.1038/s41586-023-06102-8            hybrid    
31  10.1002/ctd2.119                      gold      
32  10.1002/adfm.202411880                closed    
33  10.1002/jbm4.10217                    gold      
34  10.1016/j.echo.2018.08.016            closed    
35  10.3390/diagnostics12092063           gold      
36  10.1080/21655979.2021.2006967         gold      
37  10.1016/j.echo.2019.04.002            closed    
38  10.1006/mcne.2000.0948                closed    
39  10.1002/aet2.10924                    hybrid    
40  10.1002/smsc.202000075                gold      
41  10.1016/j.healun.2021.01.1305         bronze    
42  10.1038/s41467-021-27186-8            gold      
43  10.1038/s41467-023-39062-8            gold      
44  10.1038/s41591-024-03328-5            closed    
45  10.1093/bjs/znab101                   hybrid    
46  10.48550/arxiv.2410.01684             green     
47  10.1016/j.compscitech.2020.108140     closed    
48  10.1016/j.stem.2024.10.007            closed    
49  10.1038/s41467-019-10456-x            gold      
50  10.1038/s41590-022-01163-9            bronze    
51  10.1093/braincomms/fcaf266            gold      
52  10.1101/2025.07.02.662771             green     
53  10.1002/2327-6924.12532               closed    
54  10.1093/ejcts/ezs242                  bronze    
55  10.1021/acsami.0c06010                closed    
56  10.1002/cyto.b.22223                  bronze    
57  10.1002/mds.28059                     closed    
58  10.1016/j.healun.2019.09.002          closed    
59  10.1016/j.jacc.2010.02.048            bronze    
60  10.1038/s41591-023-02341-4            closed    
61  10.1093/cercor/bhx141                 bronze    
62  10.1101/2021.10.05.463147             green     
63  10.1101/2022.06.27.497760             green     
64  10.1101/2024.02.20.580924             green     
65  10.1161/circimaging.118.007915        bronze    
66  10.1158/2159-8290.cd-17-1246          bronze    
67  10.1186/s12909-023-04779-z            gold      
68  10.1021/acs.nanolett.2c02275          green     
69  10.1002/cncr.30921                    bronze    
70  10.1002/dvdy.22028                    bronze    
71  10.1038/s41467-025-60308-0            gold      
72  10.1038/s41586-025-08589-9            hybrid    
73  10.1038/s44318-025-00454-9            diamond   
74  10.1093/haschl/qxaf025                hybrid    
75  10.11

In [129]:
write_json('10.1016/j.jaccas.2024.102715', 'openaccess.json')

## Authors

All our publications are associated with Authors where we know:

* department
* school
* orcid
* sunet/email

In [132]:
write_json('10.1038/s41586-023-06669-2', 'author.json')

## Funders

We know funder information from OpenAlex and Dimensions.

In [140]:
pandas.read_sql("""
  SELECT publication.doi, COUNT(*) AS funders_count
  FROM publication, pub_funder_association, funder
  WHERE publication.id = pub_funder_association.publication_id
    AND pub_funder_association.funder_id = funder.id
  GROUP BY publication.doi
  HAVING COUNT(*) > 2
  ORDER BY funders_count DESC
  LIMIT 100
""", con=engine)

doi  funders_count
0   10.1117/1.nph.12.2.027801         28           
1   10.3847/1538-4365/ac78eb          24           
2   10.3847/1538-4357/acdd78          23           
3   10.1038/s41467-020-18849-z        20           
4   10.1111/all.16490                 19           
5   10.3847/1538-4357/ac6e65          18           
6   10.3847/1538-4357/ad07d0          17           
7   10.1038/s41591-024-03425-5        16           
8   10.1111/pai.13802                 16           
9   10.1016/s1470-2045(16)30214-5     16           
10  10.1093/nar/gkaf032               15           
11  10.3847/1538-4357/abe1af          15           
12  10.3847/1538-4357/ac399b          15           
13  10.1111/anae.15563                14           
14  10.1111/anae.15458                14           
15  10.1016/j.joule.2017.09.014       13           
16  10.1016/j.cell.2023.12.032        13           
17  10.1126/sciimmunol.abj2901        13           
18  10.3847/1538-3881/ac2368          13           
19  10.1038/s41586-024-08489-4        13           
20  10.3847/1538-4357/aca1c3          13           
21  10.1101/2024.01.03.24300794       13           
22  10.1101/2025.01.07.631402         13           
23  10.1021/acs.nanolett.0c04409      12           
24  10.1126/science.abe0690           12           
25  10.1021/acs.nanolett.2c04749      12           
26  10.3847/2041-8213/ac2d9a          12           
27  10.1101/2023.10.27.23297507       12           
28  10.1182/bloodadvances.2024012637  11           
29  10.1016/j.ymthe.2018.02.018       11           
30  10.1073/pnas.1607493113           11           
31  10.1016/j.cell.2022.06.008        11           
32  10.1172/jci148635                 11           
33  10.1126/scitranslmed.adc9854      11           
34  10.1002/adma.202311949            10           
35  10.1021/jacs.1c13368              10           
36  10.1038/s41467-019-08593-4        10           
37  10.1101/2025.05.20.654652         10           
38  10.1021/acs.nanolett.7b03955      10           
39  10.1038/s44286-023-00009-x        10           
40  10.1038/s41587-024-02171-w        10           
41  10.1038/s43018-022-00456-0        10           
42  10.1093/ajcn/nqac237              10           
43  10.1021/acs.jpca.5c02883          10           
44  10.1038/s41467-025-60308-0        10           
45  10.1021/acs.jpcc.2c02293          10           
46  10.3847/1538-3881/ac3492          10           
47  10.1101/2024.09.30.615861         10           
48  10.1016/j.devcel.2024.12.030      10           
49  10.1103/physrevb.103.l241108      10           
50  10.1084/jem.20210582              10           
51  10.1093/neuonc/noae112            10           
52  10.1126/science.abj2096           9            
53  10.1021/acsnano.3c05792           9            
54  10.1021/acsnano.4c12905           9            
55  10.3847/1538-4357/ace4ba          9            
56  10.1016/j.joule.2023.03.005       9            
57  10.21203/rs.3.rs-279400/v1        9            
58  10.1021/acs.jproteome.3c00474     9            
59  10.1007/s00246-021-02798-5        9            
60  10.1126/science.aao4426           9            
61  10.1101/2023.05.29.541540         9            
62  10.1002/gepi.22081                9            
63  10.1038/s41564-023-01406-x        9            
64  10.1101/2023.10.24.563664         9            
65  10.1136/ard-2022-223640           9            
66  10.1016/j.xcrm.2021.100287        9            
67  10.1183/13993003.03388-2020       9            
68  10.1093/jnci/djaf069              9            
69  10.1126/science.abj0890           9            
70  10.1038/s42003-024-05877-4        9            
71  10.1101/2023.07.31.551315         9            
72  10.1038/s41477-022-01167-1        8            
73  10.1084/jem.20182015              8            
74  10.1016/j.cell.2025.02.005        8            
75  10.1016/j.cell.2025.05.031        8            
76  10.1038/s41586-022-05277-

In [141]:
write_json('10.1038/s41467-025-60308-0', 'funders.json')